### PR TITLE
Misc HTML doc generation changes

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:

- Update doc and code examples for API calls
- Add config for HTML doc generation
- Badges and more badges!
- Add missing license file
- Fix typos
- Combine readme to module doc

### CODE REVIEW

**Comment by steffenix** [comment]
Can you rebase your changes there is a conflict.

File: lib/coingecko.ex [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
-  Hello world.
+  Returns the current price of any cryptocurrencies in any other supported
+  currencies.
 
   ## Examples
 
-      iex> Coingecko.hello()
-      :world
+      iex> Coingecko.get_simple_price(["binancecoin", "basecoin"], ["btc", "eth"])
+      {:commit,
```
Comment (steffenix) [inline lib/coingecko.ex:16]: I have abstracted the reply from the cache so it will always return {:ok ...}

Comment (kianmeng) [reply lib/coingecko.ex]: Code example updated.

**Comment by kianmeng** [comment]
> Can you rebase your changes there is a conflict.

Done.

**Pull request merged**